### PR TITLE
Fix cbw2 ros topic

### PR DESCRIPTION
**Pull request by barbalberto**
Fix errors on ROS topic message, in particular:

- measurement unit for linear joints: They were converted into radiants by mistake. Fixed
- timeStamps fot joint message: now uses time from getEncoderTime. This fixes issue on ROS TF while using Gazebo simulator. In this case both YARP & ROS clock plugin have to be instantiated.
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/barbalberto%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4061020%3Fv%3D3%22%7D%2C%20%22https%3A//github.com/drdanz%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robotology/yarp/pull/763%23issuecomment-220590054%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-05-20T12%3A14%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22Wait%20a%20bit%20to%20merge%2C%20probably%20I%27ll%20add%20one%20more%20fix%20%3B-%29%22%2C%20%22created_at%22%3A%20%222016-05-20T12%3A15%3A53Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4061020%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/barbalberto%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-05-24T08%3A40%3A30Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4061020%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/barbalberto%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222016-05-24T08%3A41%3A07Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4061020%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/barbalberto%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%2071cdc7e2d6101cc52c66651e855244606ffcc47e%20src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h%204%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64029065%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22This%20is%20another%20change%20in%20the%20ABI%20that%20could%20be%20avoided%20if%20we%20were%20using%20pimpl%20here.%5Cr%5CnI%20suggest%20that%20later%20we%20move%20all%20the%20private%20part%20of%20this%20class%20into%20an%20opaque%20pointer%20%28in%20%60devel%60%29.%22%2C%20%22created_at%22%3A%20%222016-05-20T11%3A48%3A50Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-05-24T08%3A43%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h%3AL327-334%22%7D%2C%20%22Pull%2071cdc7e2d6101cc52c66651e855244606ffcc47e%20src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64028781%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Adding%20a%20method%20to%20a%20class%20or%20changes%20the%20ABI%2C%20therefore%20according%20to%20the%20new%20workflow%2C%20this%20should%20not%20be%20done%20on%20the%20%60master%60%20branch.%5Cr%5CnAs%20discussed%20with%20%40barbalberto%2C%20next%20version%20of%20the%20patch%20will%20fix%20this%20issue%20without%20changing%20the%20ABI%2C%20and%20this%20patch%20will%20be%20applied%20on%20devel%20later%2C%20after%20this%20pr%20is%20merged.%22%2C%20%22created_at%22%3A%20%222016-05-20T11%3A47%3A16Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222016-05-24T08%3A43%3A11Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1100056%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/drdanz%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h%3AL348-356%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/robotology/yarp/pull/763%23issuecomment-220590054%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23issuecomment-220590236%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23issuecomment-221203404%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23issuecomment-221203570%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64028781%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64029065%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64349988%22%2C%20%22https%3A//github.com/robotology/yarp/pull/763%23discussion_r64349990%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/barbalberto'></a><a href='https://github.com/drdanz'></a>

- [x] <a href='#crh-comment-Pull 71cdc7e2d6101cc52c66651e855244606ffcc47e src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h 12'></a> &nbsp;<b><a href='https://github.com/robotology/yarp/pull/763#discussion_r64028781'>File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h:L348-356</a></b>
- <a href='https://github.com/drdanz'></a> Adding a method to a class or changes the ABI, therefore according to the new workflow, this should not be done on the `master` branch.
As discussed with @barbalberto, next version of the patch will fix this issue without changing the ABI, and this patch will be applied on devel later, after this pr is merged.
- [x] <a href='#crh-comment-Pull 71cdc7e2d6101cc52c66651e855244606ffcc47e src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h 4'></a> &nbsp;<b><a href='https://github.com/robotology/yarp/pull/763#discussion_r64029065'>File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h:L327-334</a></b>
- <a href='https://github.com/drdanz'></a> This is another change in the ABI that could be avoided if we were using pimpl here.
I suggest that later we move all the private part of this class into an opaque pointer (in `devel`).
- [x] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/robotology/yarp/pull/763#issuecomment-220590054'>General Comment</a></b>
- <a href='https://github.com/drdanz'></a> :shipit:
- <a href='https://github.com/barbalberto'></a> Wait a bit to merge, probably I'll add one more fix ;-)

<a href='https://www.codereviewhub.com/robotology/yarp/pull/763?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robotology/yarp/pull/763?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/robotology/yarp/pull/763?approve=0'></a>&nbsp;<a href='https://github.com/robotology/yarp/pull/763'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by drdanz** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/robotology/yarp/pull/763'>Refresh</a>.<a href='#crh-update-none'></a>

File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h [hunk 1]
```diff
@@ -347,6 +348,8 @@ class yarp::dev::ControlBoardWrapper:   public yarp::dev::DeviceDriver,
     bool ownDevices;
 #endif  //DOXYGEN_SHOULD_SKIP_THIS
 
+    bool threadInit();
```
Comment (drdanz) [inline src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h]: Adding a method to a class or changes the ABI, therefore according to the new workflow, this should not be done on the `master` branch.
As discussed with @barbalberto, next version of the patch will fix this issue without changing the ABI, and this patch will be applied on devel later, after this pr is merged.

File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h [hunk 2]
```diff
@@ -327,6 +327,7 @@ class yarp::dev::ControlBoardWrapper:   public yarp::dev::DeviceDriver,
     int               top;          // to be removed
     int               period;       // thread rate for publishing data
     bool              _verb;        // make it work and propagate to subdevice if --subdevice option is used
+    std::vector<JointTypeEnum>     jointTypes;
```
Comment (drdanz) [inline src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h]: This is another change in the ABI that could be avoided if we were using pimpl here.
I suggest that later we move all the private part of this class into an opaque pointer (in `devel`).

**Comment by drdanz** [comment]
:shipit:

**Comment by barbalberto** [comment]
Wait a bit to merge, probably I'll add one more fix ;-)

**Comment by barbalberto** [comment]
:shipit:

**Comment by barbalberto** [comment]
:shipit:

File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h (hunk 2 above)
Comment (drdanz) [inline src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h]: :+1:<a href='#crh-update-none'></a>

File: src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h (hunk 1 above)
Comment (drdanz) [inline src/libYARP_dev/src/modules/ControlBoardWrapper/ControlBoardWrapper.h]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**